app/services/customers/eu_auto_taxes_service.rb

```diff
@@ -34,16 +34,21 @@ def call
     def check_vies
       return nil if customer.tax_identification_number.blank?
 
-      response = Valvat.new(customer.tax_identification_number).exists?(detail: true)
+      # Just errors extended from Valvat::Lookup are raised, while Maintenances are not.
+      # https://github.com/yolk/valvat/blob/master/README.md#handling-of-maintenance-errors
+      # Check the Unavailable sheet per UE country.
+      # https://ec.europa.eu/taxation_customs/vies/#/help
+      response = Valvat.new(customer.tax_identification_number).exists?(detail: true, raise_error: true)
 
       after_commit { SendWebhookJob.perform_later("customer.vies_check", customer, vies_check: response.presence || error_vies_check) }
 
       response
-    rescue Valvat::RateLimitError, Valvat::Timeout, Valvat::BlockedError, Valvat::InvalidRequester => e
+    rescue Valvat::RateLimitError, Valvat::Timeout, Valvat::BlockedError, Valvat::InvalidRequester,
+      Valvat::ServiceUnavailable, Valvat::MemberStateUnavailable => e
       after_commit do
         SendWebhookJob.perform_later("customer.vies_check", customer, vies_check: error_vies_check.merge(error: e.message))
         # Enqueue a job to retry the VIES check after a delay
-        RetryViesCheckJob.set(wait: 30.seconds).perform_later(customer.id)
+        RetryViesCheckJob.set(wait: 15.minutes).perform_later(customer.id)
```

Comment:
Maybe we could keep the 30.seconds in most cases:

```suggestion
        delay = case e
        when Valvat::ServiceUnavailable, Valvat::MemberStateUnavailable
          15.minutes
        else
          30.seconds
        end
        RetryViesCheckJob.set(wait: delay).perform_later(customer.id)
```

Reply:
To be fair, I have no strong opinion on this. But since I don't know how this service is meant to behave, I'd have kept the current logic for non-maintenance errors.

I guess 5 minutes should be a good compromise.